### PR TITLE
Feature/newsletter page

### PR DESCRIPTION
**Pull request by martinyde**
https://jira.itkdev.dk/browse/HRIN-658

- Add newsletter signup iframe to specific node
- Add linkbox that points to newsletter signup page

On release:
- Add node and menu item to secondary top menu.
- Add newsletter config to local.settings that points to the new node.

### CODE REVIEW

File: web/modules/custom/hoeringsportal_config_settings/src/Form/ItkGeneralSettingsForm.php [hunk 1]
```diff
@@ -173,6 +173,27 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#default_value' => $config->get('users_manual_url'),
     ];
 
+    $node_newsletter = Node::load($config->get('node_newsletter') ?? -1);
+    $form['misc']['node_newsletter'] = [
```
Comment (rimi-itk) [inline web/modules/custom/hoeringsportal_config_settings/src/Form/ItkGeneralSettingsForm.php:177]: `newsletter_node` may be a better name (and give some sort of namespacing matching `newsletter_iframe_source` and `newsletter_iframe_height`).

A description telling that this is the page on which to show the newsletter sign-up (if the code reviewer understands the code correctly) will be useful.

File: web/themes/custom/hoeringsportal/hoeringsportal.theme [hunk 2]
```diff
@@ -329,6 +329,15 @@ function hoeringsportal_preprocess_ds_1col(&$variables) {
...
   // Check content type.
+  $itk_config = \Drupal::getContainer()->get('itk_admin.itk_config');
+  $node_newsletter = $itk_config->get('node_newsletter');
+  $variables['node_newsletter'] = $node_newsletter ?? NULL;
+  $variables['show_newsletter_signup'] = FALSE;
+  if ($node_newsletter === $variables['content']['#node']->id()) {
+    $variables['show_newsletter_signup'] = TRUE;
+    $variables['newsletter_iframe_source'] = $itk_config->get('newsletter_iframe_source');
+    $variables['newsletter_iframe_height'] = $itk_config->get('newsletter_iframe_height');
+  }
```
Comment (rimi-itk) [inline web/themes/custom/hoeringsportal/hoeringsportal.theme:340]: This should be moved above the `// Check content type.` line.

And a new comment telling that we're showing the newsletter sign-up one the selected `node_newsletter` page may be useful.

File: web/themes/custom/hoeringsportal/templates/newsletter-bottom-link.html.twig [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+          <i class="fal fa-envelope"></i>
+        </div>
+      </div>
+    </div>
+    <div class="col py-2">
+      <h5 class="newsletter-signup--header">{{ 'Would you like a direct message when deltag.aarhus.dk has new content?'|t }}</h5>
+      <div>
+        {{ 'Sign up to our message service, and be informed about you interests'|t }}
+        <span>{{ ' - ' }}</span>
+        <a href="{{ path('entity.node.canonical', {'node': node_newsletter}) }}">{{ 'Click here'|t }}</a>
```
[the remark this answers is not in the document]
Comment (martinyde) [reply web/themes/custom/hoeringsportal/templates/newsletter-bottom-link.html.twig:15]: Removed ","

The "Click here" is a design decission: 
billede

[the remark this answers is not in the document]
Comment (rimi-itk) [reply web/themes/custom/hoeringsportal/templates/newsletter-bottom-link.html.twig:15]: Who made that design decision? It's bad practice and bad for accessibility as well (cf. https://heyoka.medium.com/dont-use-click-here-f32f445d1021).

File: web/themes/custom/hoeringsportal/hoeringsportal.theme (hunk 2 above)
Comment (martinyde) [reply web/themes/custom/hoeringsportal/hoeringsportal.theme:340]: Added comment and moved code

File: web/modules/custom/hoeringsportal_config_settings/src/Form/ItkGeneralSettingsForm.php (hunk 1 above)
Comment (martinyde) [reply web/modules/custom/hoeringsportal_config_settings/src/Form/ItkGeneralSettingsForm.php:177]: Added description

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**